src/ui/class-post-submitbox.php

```diff
@@ -123,4 +153,109 @@ public function add_rewrite_and_republish_post_button( $post = null ) {
 			}
 		}
 	}
+
+	/**
+	 * Changes the 'Publish' copies in the submitbox to 'Republish' if a post is intended for republishing.
+	 *
+	 * @param string $translation The translated text.
+	 * @param string $text        The text to translate.
+	 *
+	 * @return string The to-be-used copy of the text.
+	 */
+	public function change_republish_strings_classic_editor( $translation, $text ) {
+		if ( ! $this->should_change_rewrite_republish_copy( \get_post() ) ) {
+			return $text;
+		}
+
+		if ( $text === 'Publish' ) {
+			return \__( 'Republish', 'duplicate-post' );
+		}
+
+		return $translation;
+	}
+
+	/**
+	 * Changes the 'Schedule' copy in the submitbox to 'Schedule republish' if a post is intended for republishing.
+	 *
+	 * @param string $translation The translated text.
+	 * @param string $text        The text to translate.
+	 * @param string $context     Context information for the translators.
+	 *
+	 * @return string The to-be-used copy of the text.
+	 */
+	public function change_schedule_strings_classic_editor( $translation, $text, $context ) {
+		if ( ! $this->should_change_rewrite_republish_copy( \get_post() ) ) {
+			return $text;
+		}
+
+		if ( $text === 'Schedule' ) {
+			return \__( 'Schedule republish', 'duplicate-post' );
+		}
+
+		return $translation;
+	}
+
+	/**
+	 * Changes the post-scheduled notice when a post or page intended for republishing is scheduled.
+	 *
+	 * @param array[] $messages Post updated messaged.
+	 *
+	 * @return array[] The to-be-used messages.
+	 */
+	public function change_scheduled_notice_classic_editor( $messages ) {
+		$post = \get_post();
+		if ( ! $this->should_change_rewrite_republish_copy( $post ) ) {
+			return $messages;
+		}
+
+		$permalink      = \get_permalink( $post->ID );
+		$scheduled_date = \get_the_time( \get_option( 'date_format' ), $post );
```

Comment:
According to [the design](https://www.sketch.com/s/2ec4254a-9529-4616-abe7-f6262c2ed6dd/a/Pyl8YL/play), the publish date should display also the publish _time_. See also the related test.